providers/structured/provider_test.go

```diff
@@ -171,4 +175,66 @@ var _ = Describe("bufferedLogProvider", func() {
 		}))
 	})
 
+	It("Should construct a chaining StructuredOutputLogProvider with a nil next provider", func() {
+		lp := LogProvider(nil).(*StructuredOutputLogProvider)
+		Ω(lp.LogProvider).Should(BeNil())
+	})
+
+	It("Should construct a chaining log provider with a dummy provider as next provider", func() {
+		dummyProvider := dummy.LogProvider(os.Stdout)
+		lp := LogProvider(dummyProvider).(*StructuredOutputLogProvider)
+		Ω(lp.LogProvider).Should(Equal(chaining.LogProvider(dummyProvider)))
+	})
+
+	It("Should log and then pass on the logs to the next provider", func() {
+		var (
+			msg1     = "Message 1"
+			msg2     = "Message 2"
+			details1 = "Details about message 1"
+			details2 = "Details about message 2"
+		)
+
+		// lp is `this` provider, and the `provider` is the next provider.
+		// At the end of this, they should have the same log calls, metrics and events
+		lp := LogProvider(provider).(*StructuredOutputLogProvider)
+
+		// put some logs
+		lp.Error(contextLogger, true, msg1, details1)
+		lp.Error(contextLogger, false, msg2, details2)
+		lp.Info(contextLogger, true, msg1, details1)
+		lp.Info(contextLogger, false, msg2, details2)
+		lp.Debug(contextLogger, true, msg1, details1)
+		lp.Debug(contextLogger, false, msg2, details2)
+		lp.Warn(contextLogger, true, msg1, details1)
+		lp.Warn(contextLogger, false, msg2, details2)
+
+		// put some metrics
+		lp.Record(contextLogger, log.Metrics{
+			"Metric 1": "Value 1",
+			"Metric 2": "Value 2",
+		})
+		lp.Record(contextLogger, log.Metrics{
+			"Metric 3": "Value 3",
+			"Metric 4": "Value 4",
+		})
+
+		// put some events
+		lp.RecordEvent(contextLogger, "Event1", log.Metrics{
+			"Metric 1": "Value 1",
+			"Metric 2": "Value 2",
+		})
+		lp.RecordEvent(contextLogger, "Event2", log.Metrics{
+			"Metric 3": "Value 3",
+			"Metric 4": "Value 4",
+		})
+
```

Comment:
Think you should also test that calling `lp.Wait()` causes `lp.nextProvider.Wait()` to be called. Looks like you could use dummy/provider as essentially a mock for `nextProvider`, to accomplish that verification.